client/java/src/main/java/com/linecorp/centraldogma/client/Watcher.java

```diff
@@ -178,4 +182,35 @@ default void watch(Consumer<? super T> listener) {
         requireNonNull(listener, "listener");
         watch((revision, value) -> listener.accept(value));
     }
+
+    /**
+     * @return A new {@link Watcher}, that reuses the supplied watcher and applies transformation.
+     * Note: the returned watcher is effectively filtering in a sense that,
+     * its listeners are *not* notified when a change has no effect on the transformed value.
+     * The returned watcher does not need to be closed after use.
+     */
+    default <U> Watcher<U> newChild(Function<T, U> transformer) {
+        return new TransformingWatcher<>(this, transformer);
+    }
+    /**
+     * @return A new child {@link Watcher}, which transformation is a
+     * <a href="http://tools.ietf.org/html/draft-ietf-appsawg-json-pointer-03">JSON pointer</a> query.
+     */
+    static Watcher<JsonNode> atJsonPointer(Watcher<JsonNode> watcher, String jsonPointer) {
```

Comment:
If you want to keep this method we should tell that the pointer should be [encoded](https://tools.ietf.org/html/rfc6901#section-3) in the Javadoc because we don't do that. 😉

Reply:
Good point. Added a comment